File: ios/ShortDrama/Sources/Domain/Entities/TheaterChannel.swift
import Foundation

/// Supported theater channels.
enum TheaterChannel: String, CaseIterable, Codable, Equatable, Sendable, Identifiable {
    case all
    case real
    case anime
    case movie
    case audio
    case novel
    case comic
    case bigscreen

    var id: String { rawValue }

    var title: String {
        switch self {
        case .all:
            return "找剧"
        case .real:
            return "真人"
        case .anime:
            return "动漫"
        case .movie:
            return "电影"
        case .audio:
            return "有声书"
        case .novel:
            return "小说"
        case .comic:
            return "漫画"
        case .bigscreen:
            return "大屏"
        }
    }
}

File: android/app/src/main/java/com/djs66256/short_drama/domain/model/TheaterChannel.kt
package com.djs66256.short_drama.domain.model

enum class TheaterChannel(
    val apiValue: String,
    val label: String,
) {
    ALL(apiValue = "all", label = "找剧"),
    REAL(apiValue = "real", label = "真人"),
    ANIME(apiValue = "anime", label = "动漫"),
    MOVIE(apiValue = "movie", label = "电影"),
    AUDIO(apiValue = "audio", label = "有声书"),
    NOVEL(apiValue = "novel", label = "小说"),
    COMIC(apiValue = "comic", label = "漫画"),
    BIGSCREEN(apiValue = "bigscreen", label = "大屏"),
    ;

    companion object {
        fun fromApiValue(value: String?): TheaterChannel = entries.firstOrNull {
            it.apiValue == value.orEmpty().trim()
        } ?: ALL
    }
}
